tools/vm-migrator/subnet.py

```diff
@@ -0,0 +1,243 @@
+                                                 zone=zone_3,
+                                                 maxResults=10000).execute()
+
+    mydict = []
+
+    if (result_zone_2.get("items") and zone_2):
+        result['items'] = result['items'] + result_zone_2.get("items")
+
+    if (result_zone_3.get("items") and zone_3):
+        result['items'] = result['items'] + result_zone_3.get("items")
+
+    for instances in result['items']:
+
+        headers = fields.headers
+        if ((instances['networkInterfaces'][0]['subnetwork']).endswith(subnet)):
+            csv = {
+                'name': instances['name'],
+                'id': instances['id'],
+                'machine_type': instances['machineType'],
+                'self_link': instances['selfLink'],
+                'network': instances['networkInterfaces'][0]['network'],
+                'internal_ip': instances['networkInterfaces'][0]['networkIP'],
+                'subnet': instances['networkInterfaces'][0]['subnetwork']
+            }
+
+            for i, disks in enumerate(instances['disks']):
+                if i < 9:
+                    csv['device_name_' + str(i + 1)] = disks['deviceName']
+                    csv['disk_name_' + str(i + 1)] = disk.parse_self_link(
+                        disks['source'])['name']
+                else:
+                    logging.warning(
+                        "Too many disks: dropping disk name %s with and "
+                        "device name %s" %
+                        (disk.parse_self_link(disks['source'])['name'],
+                            disks['deviceName']))
+
+            aliasIps = instances['networkInterfaces'][0].get('aliasIpRanges')
+            if (aliasIps):
+                logging.info("Found Alias IP for %s" % instances['name'])
+                for i in range(len(aliasIps)):
+                    csv['alias_ip_' + str(i + 1)] = aliasIps[i]['ipCidrRange']
+
+                    ip_name = get_alias_ip_name(
+                        project,
+                        instance.get_region_from_zone(instances['zone']),
+                        subnet, aliasIps[i]['ipCidrRange'])
+                    if (ip_name):
+                        csv['alias_ip_name_' + str(i + 1)] = ip_name
+
+                    if (aliasIps[i].get('subnetworkRangeName')):
+                        csv['range_name_' +
+                            str(i + 1)] = aliasIps[i]['subnetworkRangeName']
+
+            if (instance.is_hosted_on_sole_tenant(instances)):
+                csv['node_group'] = instance.get_node_group(instances)
+
+            mydict.append(csv)
+        with open(file_name, 'w') as csvfile:
+
+            writer = DictWriter(csvfile, fieldnames=headers)
+            writer.writeheader()
+            writer.writerows(mydict)
+
+        logging.info("Successfully written %i records to %s" %
+                     (len(mydict), file_name))
+        print("Successfully written %i records to %s" %
```

Comment:
Having print and a logger in the same place is strange. Users can choose the logging output level in the stdout if they need to.